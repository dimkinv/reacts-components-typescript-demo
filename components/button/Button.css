button {
  border: 5px solid tomato;
}
