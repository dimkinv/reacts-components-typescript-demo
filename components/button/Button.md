A simple button:

```js
<Button title="A Button" />
```