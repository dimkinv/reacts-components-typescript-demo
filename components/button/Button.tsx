import * as React from 'react';
import ButtonProps from './ButtonProps';
import ButtonState from './ButtonState';

export default class Button extends React.PureComponent<ButtonProps, ButtonState>{
    constructor(props: ButtonProps){
        super(props);
        this.state = {} as ButtonState;
    }
    render(){
        return (<button onClick={() => {console.log('clicked')}}>Hey {this.props.title}</button>)
    }
}