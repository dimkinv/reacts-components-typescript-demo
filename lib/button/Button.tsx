import * as React from 'react';
import ButtonProps from './ButtonProps';
import ButtonState from './ButtonState';

export default class Button extends React.PureComponent<ButtonProps, ButtonState>{
    constructor(props: ButtonProps){
        super(props);
        this.state = new ButtonState();
        this.props = {title: 'aa'} as ButtonProps;
    }
    render(){
        return (<button>DEMO</button>)
    }
}