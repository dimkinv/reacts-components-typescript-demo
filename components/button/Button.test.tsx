import * as React from "react";
import { render } from "enzyme";

import Button from "./Button";

describe("Button", () => {
    it("renders correctly ", () => {
      const button = render(<Button title="My Title" />);
      expect(button).toMatchSnapshot();
    });
    it("matches different titles", () => {
      ["title", "another title"].forEach((title) => {
        const button = render(<Button title={title} />);
        expect(button).toMatchSnapshot();
      })
    })
});
