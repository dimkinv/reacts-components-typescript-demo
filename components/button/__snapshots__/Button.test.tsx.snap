// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches different titles 1`] = `
<button>
  Hey title
</button>
`;

exports[`Button matches different titles 2`] = `
<button>
  Hey another title
</button>
`;

exports[`Button renders correctly  1`] = `
<button>
  Hey My Title
</button>
`;
